vmm/src/vm.rs

```diff
@@ -970,6 +990,62 @@ impl Vm {
         Ok(EntryPoint { entry_addr })
     }
 
+    #[cfg(feature = "igvm")]
+    fn load_igvm(
+        igvm: File,
+        memory_manager: Arc<Mutex<MemoryManager>>,
+        cpu_manager: Arc<Mutex<cpu::CpuManager>>,
+    ) -> Result<EntryPoint> {
+        /*
+        BIOS-e820: [mem 0x0000000000000000-0x000000000009ffff] usable
+        BIOS-e820: [mem 0x00000000000a0000-0x00000000000fffff] reserved
+        BIOS-e820: [mem 0x0000000000100000-0x00000000001fffff] ACPI data
+        BIOS-e820: [mem 0x0000000001a00000-0x0000000003c79fff] usable
+         */
+        let cur_ram_size = memory_manager.lock().unwrap().current_ram();
+        debug!("current ram size: {:?}", cur_ram_size);
+        let num_cpus = cpu_manager.lock().unwrap().vcpus().len() as u32;
+        let mut arch_mem_regions: Vec<ArchMemRegion> = Vec::new();
+        arch_mem_regions.push(ArchMemRegion {
+            base: 0x00000000e8000000,
+            size: 0x10000000,
+            r_type: RegionType::Reserved,
+        });
+        if cur_ram_size <= 3 * 1024 * 1024 * 1024 {
```

Comment:
nit: using a macro or constant for `GiB` or something similar would be a little clearer

Reply:
Fixed